### DISCUSSION TITLE
Categorical histplot not working properly with boolean data

### DISCUSSION THREAD

**Discussion by e-pet** [mwaskom/seaborn#3166] [Q&A] [answered]
Hi,

while categorical histplots generally work, they don't seem to work (properly) for boolean data.

```python
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

if __name__ == '__main__':
    df = pd.DataFrame({'y': [True, True, True, False, False]})

    plt.figure()
    # produces (undesired) float scale and the following message:
    # <__array_function__ internals>:180: RuntimeWarning: Converting input from bool to <class 'numpy.uint8'> for compatibility.
    sns.histplot(df, x='y')

    df['y'] = df['y'].map({True: 'True', False: 'False'})

    plt.figure()
    # desired output (categorical scale)
    sns.histplot(df, x='y')

    plt.show()
```

Produces the following two figures:

![image](https://user-images.githubusercontent.com/1774207/204559887-154d7748-f225-4598-9541-78ff6880f65c.png)

This is with seaborn 0.12.1.

(Unrelated, but - wouldn't it be nicer if the y axis showed integer, not float values, as these are count data?)

**Comment by mwaskom** [comment, marked as answer]
Numpy (and Python in general) upcasts boolean data to numeric dtypes where relevant (True/False are equivalent to 1/0).

**Reply by e-pet** [reply]
I am aware of this general behavior, but in this particular context it seems highly undesirable to me? Maybe I am overlooking some obvious use case, but who would ever want the current behavior?

It also seems inconsistent across seaborn; see e.g. this example from [the documentation](https://seaborn.pydata.org/tutorial/categorical.html) which doesn't print "Weekend=0.0/1.0".
```python
tips = sns.load_dataset("tips")
tips["weekend"] = tips["day"].isin(["Sat", "Sun"])
sns.catplot(
    data=tips, x="day", y="total_bill", hue="weekend",
    kind="box", dodge=False,
)
```
![image](https://user-images.githubusercontent.com/1774207/204766047-469890a4-9015-43f6-82fd-3c21f4204b79.png)
